jans-lock/cedarling/src/lock_master/mod.rs

```diff
@@ -0,0 +1,109 @@
+use std::borrow::Cow;
+use wasm_bindgen::{prelude::*, throw_val};
+use web_sys::*;
+
+use crate::{
+	http::{self, ResponseEx},
+	startup::types::PolicyStoreConfig,
+};
+
+pub(crate) mod types;
+
+#[wasm_bindgen]
+extern "C" {
+	#[wasm_bindgen(js_name = btoa)]
+	pub fn js_btoa(input: &str) -> String;
+}
+
+pub(crate) async fn init<'a>(policy_store_config: &PolicyStoreConfig) -> Cow<'a, [u8]> {
+	let PolicyStoreConfig::LockMaster {
+		url,
+		policy_store_id,
+		enable_dynamic_configuration,
+		ssa_jwt,
+		application_name,
+	} = policy_store_config
+	else {
+		unreachable!("We arrive here from the PolicyStoreConfig::LockMaster")
+	};
+
+	// Get LockMasterConfig
+	let res = http::get(&format!("{}/.well-known/lock-master-configuration", url), &[])
+		.await
+		.expect_throw("Unable to fetch LockMasterConfig from URL");
+	let lock_master_config: types::LockMasterConfig = res.into_json::<types::LockMasterConfig>().await.unwrap_throw();
+
+	// Setup dynamic SSE updates for LockMaster
+	if *enable_dynamic_configuration {
+		let sse = EventSource::new(&lock_master_config.lock_sse_uri).unwrap_throw();
+		let sse2 = sse.clone();
+
+		let onopen = Closure::once_into_js(move || {
+			let onmessage = Closure::<dyn Fn(MessageEvent)>::new(move |ev: MessageEvent| {
+				console::log_2(&JsValue::from_str("Cedarling Received message: "), &ev);
+				unimplemented!("Dynamic Configuration Updates")
+			})
+			.into_js_value();
+			sse2.set_onmessage(Some(onmessage.unchecked_ref()));
+
+			let onerror = Closure::<dyn Fn(JsValue)>::new(move |ev: JsValue| throw_val(ev)).into_js_value();
+			sse2.set_onerror(Some(onerror.unchecked_ref()));
+		});
+
+		sse.set_onopen(Some(onopen.unchecked_ref()));
+	}
+
+	// Get OAuthConfig
+	let res = http::get(&lock_master_config.oauth_as_well_known, &[]).await.expect_throw("Unable to fetch LockMasterConfig from URL");
+	let openid_config = res.into_json::<types::OAuthConfig>().await.unwrap_throw();
+
+	let client: types::OAuthDynamicClient = {
+		// OpenID dynamic client registration
+		let client_req = types::OAuthDynamicClientRequest {
+			client_name: application_name,
+			application_type: "web",
+			redirect_uris: &[],
+			token_endpoint_auth_method: "client_secret_basic",
+			software_statement: ssa_jwt,
+			contacts: &["newton@gluu.org"],
+		};
+
+		// send
+		let res = http::post(&openid_config.registration_endpoint, http::PostBody::Json(client_req), &[])
+			.await
+			.expect_throw("Unable to register client");
+		res.into_json().await.expect_throw("Unable to parse client registration response")
+	};
+
+	let grant: types::OAuthGrantResponse = {
+		// https://gluu.org/docs/gluu-server/4.0/admin-guide/oauth2/
+		let token = if let Some(client_secret) = client.client_secret {
+			format!("{}:{}", client.client_id, client_secret)
+		} else {
+			console::warn_1(&JsValue::from_str("Client Secret is not provided"));
+			client.client_id
+		};
+
+		let grant = types::OAuthGrantRequest {
+			scope: "cedarling",
```

Comment:
Incorrect scope. Please follow https://gluu.org/swagger-ui/?url=https://raw.githubusercontent.com/JanssenProject/jans/main/jans-lock/lock-master.yaml to see what scopes you'll use.

Reply:
![image](https://github.com/JanssenProject/jans/assets/53893043/07478e05-82e6-4a45-a891-0c8c78f3bf3d)

What does this mean? According to the diagram I need an Access Token with a `cedarling` scope to fetch a Policy Store from the Lock Master. Given I don't have the Lock Master's actual OpenId config, I'm not sure what scope I need to access this resource

Reply:
![image](https://github.com/JanssenProject/jans/assets/6601566/8af386be-2226-472c-b685-373a532799a7)

You'll want to request a token with these two scopes.